### PR TITLE
Update to MKL 2020.2

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Update MKL version to 2020.2 

Build number is 254 for Linux-64 and Win-64 and 258 for OSX-64.

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
@conda-forge-admin, please rerender

File: recipe/install-devel.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # not all packages have the license file.  Copy it from mkl, where we know it exists
-cp -f $SRC_DIR/mkl/info/LICENSE.txt $SRC_DIR
+cp -f $SRC_DIR/mkl/info/license.txt $SRC_DIR
```
Comment (isuruf) [inline recipe/install-devel.sh:2]:
```suggestion
cp -f $SRC_DIR/mkl/info/licenses/license.txt $SRC_DIR
```

Comment (oleksandr-pavlyk) [reply recipe/install-devel.sh]: Thank you @isuruf . I realized that. It's odd that the location only changed on unix platforms.

Comment (isuruf) [reply recipe/install-devel.sh]: The location depends on the conda-build version that the original packages were built with.

**Comment by oleksandr-pavlyk** [comment]
@isuruf It took way more commits than it should have. Should I squash?

**Comment by isuruf** [comment]
No, that's fine. Thanks

**Pull request merged**